core/src/main/scala/com/salesforce/op/dsl/RichMapFeature.scala

```diff
@@ -224,11 +225,19 @@ trait RichMapFeature {
        * tracking on the original features so it's slightly different. Fortunately, tokenization for TextMaps is done
        * via the tokenize function directly, rather than with an entire stage, so things should still work here.
        */
-      if (trackNulls) {
-        val nullIndicators = new TextMapNullEstimator[TextMap]().setInput(f +: others).getOutput()
-        new VectorsCombiner().setInput(hashedFeatures, nullIndicators).getOutput()
+      (trackTextLen, trackNulls) match {
+        case (true, true) =>
+          val textLengths = new TextMapLenEstimator[TextMap]().setInput(f +: others).getOutput()
+          val nullIndicators = new TextMapNullEstimator[TextMap]().setInput(f +: others).getOutput()
+          new VectorsCombiner().setInput(Seq(hashedFeatures, textLengths, nullIndicators): _*).getOutput()
```

Comment:
why not `.setInput(hashedFeatures, textLengths, nullIndicators)`?

Reply:
same elsewhere

Reply:
Oh no idea why I did that - fixed